src/DurableTask.SqlServer/SqlOrchestrationService.cs

```diff
@@ -570,19 +571,7 @@ public override async Task SendTaskOrchestrationMessageAsync(TaskMessage message
                     return state;
                 }
 
-                try
-                {
-                    await Task.Delay(TimeSpan.FromSeconds(1), combinedCts.Token);
-                }
-                catch (TaskCanceledException)
-                {
-                    if (timeoutCts.Token.IsCancellationRequested)
-                    {
-                        throw new TimeoutException($"A caller-specified timeout of {timeout} has expired, but instance '{instanceId}' is still in an {state?.OrchestrationStatus.ToString() ?? "unknown"} state.");
-                    }
-
-                    throw;
-                }
+                await Task.Delay(TimeSpan.FromSeconds(1), combinedCts.Token);
```

Comment:
The above code was deleted to ensure that we only throw `OperationCanceledException` (or `TaskCanceledException`) if this times out. Otherwise, sometimes we'd throw `TimeoutException` and sometimes we'd throw `OperationCanceledException`, depending on which line of code was running when the cancellation token was signaled.